client/lib/reader-lists-tags/store.js

```diff
@@ -0,0 +1,94 @@
+// Reader Lists Tag Subscription Store
+
+// External dependencies
+import { OrderedSet, fromJS } from 'immutable';
+
+// Internal dependencies
+import { action as actionTypes } from 'lib/reader-lists-tags/constants';
+import { createReducerStore } from 'lib/store';
+
+const initialState = {
+	tags: OrderedSet(), // eslint-disable-line new-cap
+	errors: [],
+	currentPage: {},
+	isLastPage: {},
+	isFetching: false
+};
+
+const ReaderListsTagsStore = createReducerStore( ( state, payload ) => {
+	switch ( payload.action.type ) {
+		case actionTypes.ACTION_RECEIVE_READER_LIST_TAGS:
+			return ReaderListsTagsStore.receiveTags( payload.action.data );
+
+		case actionTypes.ACTION_RECEIVE_READER_LIST_TAGS_ERROR:
+			const errors = state.get( 'errors' );
+			return state.set( 'errors', errors.push( payload.action.error ) );
+
+		case actionTypes.ACTION_FETCH_READER_LIST_TAGS:
+			return state.set( 'isFetching', true );
+
+		case actionTypes.ACTION_FETCH_READER_LIST_TAGS_COMPLETE:
+			return state.set( 'isFetching', false );
+	}
+
+	return state;
+}, fromJS( initialState ) );
+
+ReaderListsTagsStore.getTagsForList = function( listId ) {
+	const state = ReaderListsTagsStore.get();
+	return state.get( 'tags' ).filter( function( tag ) {
+		return tag.get( 'list_ID' ) === parseInt( listId );
+	} ).toJS();
```

Comment:
really don't want to call toJS on these things. it's really expensive and invalidates anything that's trying to be a pure component